python/dgl/nn/pytorch/graph_transformer.py

```diff
@@ -298,41 +302,276 @@ def forward(self, g, edge_feat):
+    :math:`r_i` is the Cartesian coordinate of atom :math:`i`,
+    :math:`\gamma_{(i,j)}, \beta_{(i,j)}` are learnable scaling factors of
+    the :math:`k`-th Gaussian Basis Kernel.
+
+    Parameters
+    ----------
+    num_kernels : int
+        Number of Gaussian Basis Kernels to be applied.
+        Each Gaussian Basis Kernel contains a learnable kernel center
+        and a learnable scaling factor.
+    num_heads : int, optional
+        Number of attention heads if multi-head attention mechanism is applied.
+        Default : 1.
+    max_node_type : int, optional
+        Maximum element in :attr:`node_type`,
+        which should be set as 0 if and only if :attr:`node_type` is None.
+        Default : 1.
+
+    Examples
+    --------
+    >>> import torch as th
+    >>> import dgl
+    >>> from dgl.nn import SpatialEncoder3d
+
+    >>> u = th.tensor([0, 0, 0, 1, 1, 2, 3, 3])
+    >>> v = th.tensor([1, 2, 3, 0, 3, 0, 0, 1])
+    >>> g = dgl.graph((u, v))
+    >>> coordinate = th.rand(4, 3)
+    >>> node_type = th.tensor([7, 3, 3, 2])
+    >>> spatial_encoder = SpatialEncoder3d(num_kernels=4,
+    ...                                    num_heads=8,
+    ...                                    max_node_type=7)
+    >>> out = spatial_encoder(g, coordinate, node_type=node_type)
+    >>> print(out.shape)
+    torch.Size([1, 4, 4, 8])
+    """
+
+    def __init__(self, num_kernels, num_heads=1, max_node_type=0):
+        super().__init__()
+        self.num_kernels = num_kernels
+        self.num_heads = num_heads
+        self.max_node_type = max_node_type
+        self.gaussian_means = nn.Embedding(1, num_kernels)
+        self.gaussian_stds = nn.Embedding(1, num_kernels)
+        self.linear_layer_1 = nn.Linear(num_kernels, num_kernels)
+        self.linear_layer_2 = nn.Linear(num_kernels, num_heads)
+        if max_node_type == 0:
+            self.mul = nn.Embedding(1, 1)
+            self.bias = nn.Embedding(1, 1)
+        else:
+            self.mul = nn.Embedding(max_node_type + 1, 2)
+            self.bias = nn.Embedding(max_node_type + 1, 2)
+
+        nn.init.uniform_(self.gaussian_means.weight, 0, 3)
+        nn.init.uniform_(self.gaussian_stds.weight, 0, 3)
+        nn.init.constant_(self.mul.weight, 0)
+        nn.init.constant_(self.bias.weight, 1)
+
+    def forward(self, g, coord, node_type=None):
+        """
+        Parameters
+        ----------
+        g : DGLGraph
+            A DGLGraph to be encoded, which must be a homogeneous one.
+        coord : th.Tensor
+            3D coordinates of nodes in :attr:`g`,
+            of shape :math:`(N, 3)`,
+            where :math:`N`: is the number of nodes in :attr:`g`.
+        node_type : th.Tensor, optional
+            Type of nodes in :attr:`g`, of shape :math:`(N)`
+            indexed to assign learnable parameter
+            :math:`\gamma_{(i,j)}, \beta_{(i,j)}`,
+            which would not vary if :attr:`node_type` is None.
+            Default : None.

```

Comment:
```suggestion
        node_type : torch.Tensor, optional
            Node types of :attr:`g`. Default: None.
            * if :attr:`max_node_type` is 1. :attr:`node_type` needs to be a tensor in
              shape :math:`(N,)`. The scaling factors of each pair of nodes are determined
              by their node types.
            * Otherwise, :attr:`node_type` should be None.
```